tools/generator/autorest_ext/generation.go

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package autorest_ext
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/ahmetb/go-linq/v3"
+
+	"github.com/Azure/azure-sdk-for-go/tools/generator/autorest"
+	"github.com/Azure/azure-sdk-for-go/tools/generator/autorest/model"
+	"github.com/Azure/azure-sdk-for-go/tools/generator/sdk"
+	"github.com/Azure/azure-sdk-for-go/tools/generator/utils"
+	"github.com/Azure/azure-sdk-for-go/tools/internal/exports"
+	sdkutils "github.com/Azure/azure-sdk-for-go/tools/internal/utils"
+)
+
+type GenerateContext interface {
+	SDKRoot() string
+	SpecRoot() string
+	RepoContent() map[string]exports.Content
+}
+
+type GenerateInput struct {
+	// Readme is the relative path of the readme file to the root directory of azure-sdk-for-go
+	Readme string
+	// Tag is the readme tag to be generated
+	Tag string
+	// SDKRoot is the root directory of azure-sdk-for-go
+	SDKRoot string
+	// CommitHash is the head commit hash of azure-rest-api-specs
+	CommitHash string
+	// Options specifies the options that this generation task will be using
+	Options model.Options
+}
+
+type GenerateOptions struct {
+	// MetadataOutputRoot specifies the root directory of all the metadata goes.
+	// Metadata will be generated to a temp directory if not specified.
+	// The metadataOutput directory will not be removed after the generation succeeded
+	MetadataOutputRoot string
+	// Stderr ...
+	Stderr io.Writer
+	// Stdout ...
+	Stdout io.Writer
+}
+
+type GenerateResult struct {
+	MetadataOutputRoot string
+	Metadata           autorest.GenerationMetadata
+	Package            autorest.ChangelogResult
+}
+
+func GeneratePackage(ctx GenerateContext, input GenerateInput, options GenerateOptions) (*GenerateResult, error) {
```

Comment:
Package `tools/generator/autorest` already has the same function, please use that instead

Reply:
fixed